commons-shared/src/main/scala/com/avsystem/commons/collection/CollectionAliases.scala

```diff
@@ -14,103 +14,103 @@ import scala.{collection => sc}
   */
 trait CollectionAliases {
   type BTraversable[+A] = sc.Traversable[A]
-  final val BTraversable = sc.Traversable
+  final lazy val BTraversable = sc.Traversable
```

Comment:
That's right, `lazy val` generates much more code than `val`. On the other hand `lazy val` can be removed in DCE phase. These aliases (mixed in `commons` package object) force compiler to include unused parts of scala collections into JS code. 
We can create different versions of `commons` package object for JVM and JS, but I'm afraid IntelliJ cannot understand it.

Reply:
I see.

I'd prefer to refactor `commons` package object and remove some stuff from it. It will cause incompatibilities, but I still think it's less ugly solution. Perhaps it was a mistake to put all this stuff into the `commons` package by default.

As you said, `val`s can't be DCEd by default because they may have hidden side effects.
I wonder if ScalaJS guys noticed this problem and maybe they're planning to do something about it.

Reply:
Hmm... maybe we could simply change them into `final def`s?

```scala
final def BTraversable: sc.Traversable.type = sc.Traversable
```

This makes them no longer _stable identifiers_ (e.g. you can't do things like `BTraversable.type`) but I can't think of any use case where it might be needed.

However, this solution does not apply to `val`s in `GenCodec` because there I actually want to ensure single instance.

Reply:
It's even better, if we don't need stable identifiers.